src/peft/peft_model.py

```diff
@@ -1137,6 +1151,20 @@ def load_adapter(
         load_result = set_peft_model_state_dict(
             self, adapters_weights, adapter_name=adapter_name, ignore_mismatched_sizes=ignore_mismatched_sizes
         )
+        missing_keys, unexpected_keys = load_result.missing_keys, load_result.unexpected_keys
+        tuner = self.peft_config[adapter_name].peft_type
+
+        if unexpected_keys:
```

Comment:
In a first step, I would suggest not making any warnings or print statements here, since this reflects the current state of things. Instead, I would suggest that we only determine the `adapter_missing_keys` and then override the `load_results.missing_keys` with the new value. WDYT?

Reply:
Makes sense- overwriting the missing keys should work. Incase adapter is not loaded perfectly then the user can inspect  the `load_result`.